processing/src/main/java/org/apache/druid/segment/join/JoinableFactoryWrapper.java

```diff
@@ -177,7 +178,10 @@ static JoinClauseToFilterConversion convertJoinToFilter(
         }
         return new JoinClauseToFilterConversion(null, false);
       }
-      final Filter onlyFilter = new InDimFilter(leftColumn, columnValuesWithUniqueFlag.getColumnValues());
+      final Filter onlyFilter = Filters.toFilter(new InDimFilter(
```

Comment:
It is worth adding a test for this

Reply:
@rohangarg this was missed during the refactoring of https://github.com/apache/druid/pull/14237. There are tests already in place

Reply:
Added a test to preserve this behavior for future